1SirFishy Onboarding

kepler452b123 opened the pull request:
# Purpose
I read instructions and executed instructions

# New Changes
Nothing new, except the style

# Testing
Absolutely none. Fully confident it works

# Outstanding Changes
N.

kepler452b123 committed:
"Added device address of LM75BD using the
data sheet and also started working on the
function that reads temperatures from
LM75BD. Positive readings work, but tests say that
the address and negative readings are incorrect."

kepler452b123 committed:
Passes all 4 unit tests, reading from the temperat
ure register of the driver is successful. Also
successfully converts the readings to celsius

kepler452b123 committed:
Completed OS Handler

kepler452b123 committed:
Changed buffers from dynamically allocated memory
to static in the driver read function

kepler452b123 committed:
DID EPIC STUFF (REALIZED ARRAYS ARE ALREADY
JUST POINTERS TO FIRST INDEX, NOW PASSING buf1 and buf2
TO I2CSEND AND I2CRECEIVE INSTEAD OF &buf1 AND &buf2) TO
FIX MEANINGLESS TESTING ERRORS

Navtajh04 commented on lm75bd/lm75bd.c:
```diff
@@ -27,8 +27,34 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
-  /* Implement this driver function */
+  uint8_t buf1[1];
+  uint8_t buf2[2];
+  buf1[0] = 0b00000000;
```
just do `buf1[0] = {0};` for readability

Navtajh04 replied:
maybe make these names more descriptive as well

Navtajh04 commented on lm75bd/lm75bd.c:
```diff
@@ -27,8 +27,34 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
 }
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
-  /* Implement this driver function */
```
check if temp is NULL

Navtajh04 replied:
I dont think this was addressed

Navtajh04 commented on lm75bd/lm75bd.c:
```diff
@@ -27,8 +27,34 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
+  buf1[0] = 0b00000000;
+  i2cSendTo(devAddr, buf1 , 1);
+  i2cReceiveFrom(devAddr, buf2, 2);
+  uint16_t tempReading = 0b0000000000000000;
```
just use 0 for readabity since 0 = 0b0000000000000000

Navtajh04 commented on lm75bd/lm75bd.c:
```diff
@@ -27,8 +27,34 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
+  )*/
+  tempReading = tempReading | buf2[0];
+  tempReading = tempReading << 8;
+  tempReading = tempReading | buf2[1];
```
this can be done in one step instead of splitting it up into 3 just do `tempReading |= (buf2[0] << 8) | buf2[1];`

Navtajh04 commented on lm75bd/lm75bd.c:
```diff
@@ -27,8 +27,34 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
+  {
+    tempReading  = tempReading ^ 0b0000011111111111;
+    *temp = (float)((-(tempReading + 1.0)) / 8.0);
+  }
```
if you make `tempReading` and int you don't need this you can just do `*temp = (float)tempReading*0.125`

Navtajh04 commented on services/thermal_mgr/thermal_mgr.c:
```diff
@@ -42,19 +42,44 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
```
check if the queue has been created already

Navtajh04 commented on services/thermal_mgr/thermal_mgr.c:
```diff
@@ -42,19 +42,44 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
```
check if event is NULL

Navtajh04 commented on services/thermal_mgr/thermal_mgr.c:
```diff
@@ -42,19 +42,44 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
+  xQueueSend(thermalMgrQueueHandle, event, 0);
```
check the return value since this function would fail if the queue is full

Navtajh04 commented on services/thermal_mgr/thermal_mgr.c:
```diff
@@ -42,19 +42,44 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  thermal_mgr_event_t event;
+  event.type = THERMAL_MGR_EVENT_CHECK_FOR_INTPT;
```
if this function runs, then an interrupt has happened so I would rename this

Navtajh04 commented on services/thermal_mgr/thermal_mgr.c:
```diff
@@ -42,19 +42,44 @@ void initThermalSystemManager(lm75bd_config_t *config) {
+  thermal_mgr_event_t event;
   while (1) {
-    
+    if (xQueueReceive(thermalMgrQueueHandle, &event, 0))
```
since we are only expecting to receive events every 5 seconds, you can set the block time to be longer to avoid wasting CPU time by repeatedly polling the queue

Navtajh04 commented on services/thermal_mgr/thermal_mgr.c:
```diff
@@ -42,19 +42,44 @@ void initThermalSystemManager(lm75bd_config_t *config) {
+      if (event.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD)
+      {
+        float temp;
+        readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp);
```
check the return value to see if the function failed

Navtajh04 commented on services/thermal_mgr/thermal_mgr.c:
```diff
@@ -42,19 +42,44 @@ void initThermalSystemManager(lm75bd_config_t *config) {
+      else if (event.type == THERMAL_MGR_EVENT_CHECK_FOR_INTPT)
+      {
+        float temp;
+        readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp);
```
check the return value to see if the function failed

Navtajh04 commented on services/thermal_mgr/thermal_mgr.c:
```diff
@@ -42,19 +42,44 @@ void initThermalSystemManager(lm75bd_config_t *config) {
+        {
+          overTemperatureDetected();
+        }
+        if (temp <= 75)
```
you can just do else since if its not > 75 then it will be <= 75

Navtajh04 commented on lm75bd/lm75bd.c:
```diff
@@ -27,8 +27,34 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
+  uint8_t buf1[1];
+  uint8_t buf2[2];
+  buf1[0] = 0b00000000;
+  i2cSendTo(devAddr, buf1 , 1);
```
check the return value to see if the function failed

Navtajh04 replied:
Use the provided logging macros

Navtajh04 commented on lm75bd/lm75bd.c:
```diff
@@ -27,8 +27,34 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
+  uint8_t buf2[2];
+  buf1[0] = 0b00000000;
+  i2cSendTo(devAddr, buf1 , 1);
+  i2cReceiveFrom(devAddr, buf2, 2);
```
check the return value to see if the function failed

kepler452b123 committed:
Increased readability by compiling steps and
changing binary representations to integers in driver
functions. Also included error checks, as well as input
validations for pointer arguments. Did not use error functions
such as LOG_IF_ERROR_CODE(functionToCall()) and
RETURN_IF_ERROR_CODE(functionToCall()), as these were
added to the main repo after I had initially forked
and cloned the repo.

kepler452b123 committed:
Added one additional check to see if queue is made
in thermal_mgr function

Navtajh04 commented on lm75bd/lm75bd.c:
```diff
@@ -27,8 +27,43 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
-  /* Implement this driver function */
+  uint8_t bufRead[1];
+  uint8_t bufWrite[2];
+  bufRead[0] = 0;
```
just do `uint8_t bufRead[1] = {0}` when you initialize the array

Navtajh04 commented on lm75bd/lm75bd.c:
```diff
@@ -27,8 +27,43 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
+  uint8_t bufRead[1];
+  uint8_t bufWrite[2];
+  bufRead[0] = 0;
+  if (temp == NULL)
+  {
+    return ERR_CODE_INVALID_ARG;
+  }
```
generally best to validate the parameters at the top of your function before you do anything else

Navtajh04 commented on lm75bd/lm75bd.c:
```diff
@@ -27,8 +27,43 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
+  {
+    return ERR_CODE_INVALID_ARG;
+  }
+  error_code_t sendResult = i2cSendTo(devAddr, bufRead , 1);
+  if (sendResult != ERR_CODE_SUCCESS)
+  {
+    return sendResult;
+  }
+  error_code_t receiveResult = i2cReceiveFrom(devAddr,bufWrite, 2);
+  if (receiveResult != ERR_CODE_SUCCESS)
+  {
+    return receiveResult;
+  }
```
just use `RETURN_IF_ERROR_CODE` so that these errors gets logged as well

Navtajh04 reviewed (changes requested):
just a couple more things

kepler452b123 committed:
Merge pull request #1 from UWOrbital/level3

Add logging macros (#92)

kepler452b123 committed:
Used Logging macros, shifted some shit around

Navtajh04 reviewed (commented):
LGTM
